x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -24,6 +25,10 @@ import { CloudSecurityPostureConfig } from '../config';
 interface IndexTemplateSettings {
   index: {
     default_pipeline: string;
+    codec?: string;
+    mapping?: {
+      ignore_malformed: boolean;
+    };
```

Comment:
@joshdover, why those 2 new params are requires?

Reply:
These attributes were already being supplied previously by Fleet in the index template you were copying. With this PR, they're not longer added directly to the index template and instead are sourced from the `logs@settings` component template.

Since I'm filtering that component template out as mentioned above, I added these settings back explicitly to ensure that this index gets the same settings as it did before this change. I'll leave it to your team to decide if you need these settings or not, but they generally should help reduce storage costs and not drop any data.